File: sdks/flutter/lib/src/alias/alias_client.dart
import '../astro_http.dart';
import '../models/models.dart';

class AliasClient {
  final AstroHttpClient _http;
  const AliasClient(this._http);

  Future<AliasProfile> getProfile(String alias) async {
    final json = await _http.get('/alias/$alias');
    return AliasProfile.fromJson(json as Map<String, dynamic>);
  }

  Future<List<LinkedAccount>> getAccounts(String alias) async {
    final json = await _http.get('/alias/$alias/accounts');
    final list = (json as Map<String, dynamic>)['accounts'] as List;
    return list.map((e) => LinkedAccount.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<AliasProfile> deactivate(String alias) async {
    final json = await _http.post('/alias/$alias/deactivate');
    return AliasProfile.fromJson(json as Map<String, dynamic>);
  }

  Future<ResolveResult> resolve(String alias) async {
    final json = await _http.get('/alias/$alias/resolve');
    return ResolveResult.fromJson(json as Map<String, dynamic>);
  }
}

File: sdks/swift/Sources/AstroSDK/Alias/AliasClient.swift
import Foundation

public final class AliasClient {
    private let http: AstroHTTPClient
    init(http: AstroHTTPClient) { self.http = http }

    public func getProfile(_ alias: String) async throws -> AliasProfile {
        try await http.perform(http.request(http.url("/alias/\(alias)")))
    }

    public func getAccounts(_ alias: String) async throws -> [LinkedAccount] {
        let result = try await http.perform(http.request(http.url("/alias/\(alias)/accounts"))) as AccountsWrapper
        return result.accounts
    }

    public func resolve(_ alias: String) async throws -> ResolveResult {
        try await http.perform(http.request(http.url("/alias/\(alias)/resolve")))
    }

    public func deactivate(_ alias: String) async throws -> AliasProfile {
        try await http.perform(http.request(http.url("/alias/\(alias)/deactivate"), method: "POST"))
    }
}

private struct AccountsWrapper: Decodable { let accounts: [LinkedAccount] }
